Info-Orbs/src/model/webDataModel.cpp

```diff
@@ -0,0 +1,188 @@
+#include "model/webDataModel.h"
+
+String WebDataModel::getLabel() {
+    return m_label;
+}
+
+void WebDataModel::setLabel(String label) {
+    if (m_label != label) {
+        m_label = label;
+        m_changed = true;
+    }
+}
+
+String WebDataModel::getData() {
+    return m_data;
+}
+
+void WebDataModel::setData(String data, int32_t defaultColor, int32_t defaultBackground) {
+    if (m_data != data) {
+        m_data = data;
+        if (data[0] == '[') {
+            // Serial.println("JSON");
+            JsonDocument doc;
+            DeserializationError error = deserializeJson(doc, data);
+            setElementsCount(doc.size());
+            for (int i = 0; i < doc.size(); i++) {
+                WebDataElementModel element = WebDataElementModel();
+                element.parseData(doc[i].as<JsonObject>(), defaultColor, defaultBackground);
+                m_elements[i] = element;
+            }
+        } else {
+            setElementsCount(0);
+        }
+        m_changed = true;
```

Comment:
If we keep an `m_changed` at this level, the individual elements don't really need it,
and the getter / setters could be removed?

Reply:
yes, that m_changes should be moved into the else above as it is intended for the old version where data is a single value.

Reply:
but on second thought maybe I need to put some more thought into changed state with regard to the individual element models. As I don't think it is being used right now but it should be. For now I am going to leave this in place and maybe it will stay but I will add checks on the draw method for each element too. So this will mark the wholeDataModel as in need of redrawing which means it will look at each individual element to determine if it needs to be redrawn.Or maybe we will stick to redrawing the entire screen if there any changes. Either way it should be resolved in a future PR.